core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -95,55 +103,85 @@ func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context,
 	ctx, cancel := context.WithTimeout(ctx, timeoutDuration+margin)
 	defer cancel()
 
-	payload, err := json.Marshal(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal fetch request: %w", err)
+	operation := c.toRetryableFunc(ctx, messageID, req)
+
+	if req.WithRetries == nil || !*req.WithRetries {
+		return operation()
 	}
 
-	ch, err := c.responses.new(messageID)
-	if err != nil {
-		return nil, fmt.Errorf("duplicate message received for ID: %s", messageID)
+	maxTries := uint(defaultMaxTries)
+	if req.MaxTries != nil && *req.MaxTries <= defaultMaxTries {
```

Comment:
As a workflow author I could set this to be 1,000,000 retries.

I understand that there is a max time DefaultMaxElapsedTime, which is currently 15 mins (which seems high). So eventually the request would time out before reaching this number.

I'd argue for a reasonable upper bound here that is validated when the a workflow is created.